engine/src/test/java/org/kigalisim/engine/state/EngineUnitStateTest.java

```diff
@@ -0,0 +1,345 @@
+/**
+ * Unit tests for engine unit state classes.
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.kigalisim.engine.state;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.kigalisim.engine.Engine;
+import org.kigalisim.engine.number.EngineNumber;
+import org.kigalisim.engine.number.UnitConverter;
+
+/**
+ * Unit tests for the ConverterStateGetter and OverridingConverterStateGetter classes.
+ */
+public class EngineUnitStateTest {
+
+  /**
+   * Mock inner StateGetter implementation for testing.
+   */
+  private static class MockInnerStateGetter implements StateGetter {
```

Comment:
Please replace this with Mockito and, if no longer needed, please remove MockInnerStateGetter